lib/widgets/investigation_details.dart

```diff
@@ -0,0 +1,127 @@
+import 'package:connect2bahmni/domain/models/omrs_order.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import '../utils/debouncer.dart';
+
+class InvestigationDetails extends StatefulWidget {
+  final OmrsOrder investigation;
+  const InvestigationDetails({super.key, required this.investigation});
+
+  @override
+  State<InvestigationDetails> createState() => _InvestigationDetailsState();
+}
+
+class _InvestigationDetailsState extends State<InvestigationDetails> {
+  OmrsOrder _omrsOrder = OmrsOrder();
+  final TextEditingController _notesController = TextEditingController();
+  final Debouncer _debouncer = Debouncer();
+  OmrsOrderType? _selectedOrderType;
+@override
+  void initState(){
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var arg = ModalRoute.of(context)!.settings.arguments;
+    if (arg is OmrsOrder) {
+      _omrsOrder = arg;
+    } else {
+      _omrsOrder = widget.investigation;
+    }
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Add Investigation"),
+      ),
+      body: Container(
+        child: Column(
+          children: [
+            SizedBox(
+              height: 20,
+            ),
+            _investigationDisplay(),
+            SizedBox(
+              height: 20,
+            ),
+            _commentToFulfiller(),
+            ElevatedButton(
+              style: ButtonStyle(
+                  shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                      RoundedRectangleBorder(
+                          borderRadius: BorderRadius.circular(10.0),
+                          side: const BorderSide(color: Colors.red)
+                      )
+                  )
+              ),
+              onPressed: () {
+                _debouncer.stop();
+                _omrsOrder.commentToFulfiller = _notesController.text;
+                Navigator.pop(context, _omrsOrder);
+              },
+              child: const Text('Add'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _investigationDisplay() {
+    String? display = _omrsOrder.concept?.display;
+    display ??= '??';
+    return Container(
+      child: Text(display,style: TextStyle(
+        fontSize: 20
+      ),),
+    );
+  }
+  Widget _commentToFulfiller(){
+    return Expanded(
+      child: TextFormField(
```

Comment:
You donot need a FormField as you do not have a form. So you can just use a simple text field

Reply:
Got it